python/ray/data/_internal/block_batching.py

```diff
@@ -183,8 +184,10 @@ def _sliding_window(iterable: Iterable, n: int, clear_block_after_read: bool = F
         yield tuple(window)
     for elem in it:
         block_ref = window.popleft()
-        if clear_block_after_read:
-            ray._private.internal_api.free(block_ref, local_only=False)
+        if clear_block_after_read and DatasetContext.get_current().eager_free:
+            trace_deallocation(block_ref, "block_batching._sliding_window")
```

Comment:
Since this tracing utility is for debugging only, why is this also used on prod path?

Reply:
The call is always there, but it's a no-op when the flag is disabled.